astropy/visualization/wcsaxes/axislabels.py

```diff
@@ -90,15 +100,45 @@ def draw(
 
             padding = text_size * self.get_minpad(axis)
 
+            loc = self.get_loc(axis)
+            if axis in "tbhc":
+                loc = loc if loc is not None else rcParams["xaxis.labellocation"]
+                _api.check_in_list(("left", "center", "right"), loc=loc)
+
+                bary = {
+                    "left": 0,
+                    "center": 0.5,
+                    "right": 1,
+                }[loc]
+            elif axis in "lrv":
+                loc = loc if loc is not None else rcParams["yaxis.labellocation"]
+                _api.check_in_list(("bottom", "center", "top"), loc=loc)
+
+                bary, loc = {
+                    "bottom": (0, "right"),
+                    "center": (0.5, "center"),
+                    "top": (1, "left"),
+                }[loc]
+            else:
+                if loc != "center":
+                    warnings.warn(
+                        f"Only loc = 'center' is implemented at the moment for axis '{axis}'"
+                    )
+                loc = "center"
+                bary = 0.5
+
             # Find position of the axis label. For now we pick the mid-point
             # along the path but in future we could allow this to be a
             # parameter.
-            x, y, normal_angle = self._frame[axis]._halfway_x_y_angle()
+            x, y, normal_angle = self._frame[axis]._barycentric_x_y_angle(bary)
```

Comment:
this looks orthogonal to what you're trying to achieve in this PR, and I suspect this could be what's breaking existing image tests. Can you explain (or possibly, revert ?) ?